### PR TITLE
1.16.4 - Full Litematica Building Support

### PR DESCRIPTION
**Pull request by EmersonDove**
Added full Litematica support. Usage is ```#build <filepath>```.

### CODE REVIEW

File: src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java [hunk 1]
```diff
@@ -67,6 +68,24 @@ public IStaticSchematic parse(InputStream input) throws IOException {
                     throw new UnsupportedOperationException("Unsupported Version of a Sponge Schematic");
             }
         }
+    },
+    /**
+     * The Litematica Schematic Specification. Commonly denoted by the ".litematic" file extension.
+     *
```
Comment (l1ving) [inline src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java:74]: no spec in the java doc?

**Review by l1ving — APPROVED** [review]
I have a lot of gripes with the style but that's up to leijurv

File: src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java (hunk 1 above)
Comment (EmersonDove) [reply src/main/java/baritone/utils/schematic/format/DefaultSchematicFormats.java:74]: Litematica has no official specification page I could link to. I opted to reuse the MCEdit comment (I think that's what you mean).

**Comment by ZacSharp** [comment]
Answering https://github.com/cabaletta/baritone/issues/2363#issuecomment-775638133
I think this should be on the first version you can do it on.

**Comment by kawaii** [comment]
It's worth noting as a possible extension to this that Baritone currently features a `#schematica` command, which builds the currently open schematic file (in the hologram location I assume), would be nice to support that with all of the supported schematic types too.

**Comment by calloutplayed** [comment]
I pefer Update the full file

**Comment by Colonizor48** [comment]
just add this please.

**Comment by AngelFireLA** [comment]
pls merge

**Comment by AngelFireLA** [comment]
pls merge

**Comment by scorbett123** [comment]
Don't spam it will get you no-where.

**Comment by scorbett123** [comment]
@leijurv Is it possible that you can look at this? I think that it is important. Thanks.

**Comment by ehylo** [comment]
Shouldn't this be on the master branch instead of 1.16.4? or am I just dumb?

**Comment by kal-MenKissing** [comment]
Wait ehylo? Funny seeing you here and yes it should be merged into master branch during update time

**Comment by scorbett123** [comment]
> Shouldn't this be on the master branch instead of 1.16.4? or am I just dumb?

yea probably, someone just has to port it to 1.12, didn't notice that.

@EmersonDove can you recreate this pull based on master.

**Comment by EmersonDove** [comment]
> > Shouldn't this be on the master branch instead of 1.16.4? or am I just dumb?
> 
> yea probably, someone just has to port it to 1.12, didn't notice that.
> 
> @EmersonDove can you recreate this pull based on master.

Sure, I'll leave this one open until I have the next one up.

**Comment by scorbett123** [comment]
Thanks.

**Comment by StoneCommander** [comment]
#2544 
New pull.

**Pull request closed without merging**